chain/client/src/sync/state.rs

```diff
@@ -378,6 +397,39 @@ impl StateSync {
         }
         let split_states = epoch_manager.will_shard_layout_change(&prev_hash)?;
 
+        for msg in self.state_parts_mpsc_rx.try_iter() {
+            let shard_id = msg.shard_id;
+            let part_id = msg.part_id.idx;
+            if msg.sync_hash != sync_hash {
+                debug!(target: "sync",
+                    "Recieved message for other sync hash: shard_id {}, part_id {} expected sync_hash {} recieved sync_hash {}.",
+                    &shard_id,
+                    &part_id,
+                    &sync_hash,
+                    &msg.sync_hash
+                );
+                continue;
+            }
+            info!(target: "sync", "downloaded: shard_id {}, part_id {}.", shard_id, part_id );
+            if let Some(shard_sync_download) = new_shard_sync.get_mut(&shard_id) {
+                if shard_sync_download.status != ShardSyncStatus::StateDownloadParts {
+                    continue;
+                }
+                if let Some(part_download) = shard_sync_download.downloads.get_mut(part_id as usize)
+                {
+                    update_sync_status |= check_external_storage_part_response(
```

Comment:
The function can now be renamed to something more fitting. It's no longer checking, because the response is definitely available.

Reply:
renamed to `process_part_response`